src/complex/DataStructure/Factory/NeighborListFactory.hpp

```diff
@@ -33,8 +67,11 @@ class COMPLEX_EXPORT NeighborListFactory : public IDataFactory
    * to create the generated DataObject under.
    * @return H5::ErrorType
    */
-  H5::ErrorType readH5Group(H5::DataStructureReader& dataStructureReader, const H5::GroupReader& parentReader, const H5::GroupReader& groupReader,
-                            const std::optional<DataObject::IdType>& parentId = {}, bool preflight = false) override;
+  H5::ErrorType readH5Group(H5::DataStructureReader& dataStructureReader, const H5::GroupReader& parentReader, const H5::GroupReader& groupReader, const std::optional<DataObject::IdType>& parentId,
+                            bool preflight) override
```

Comment:
@bpenniebq Did we purposely get rid of the defaults for parentId and preflight in this function and readH5Dataset down below?